pages/index.tsx

```diff
@@ -1,7 +1,114 @@
-import React, { Component } from 'react'
+import Contest from '../components/Contest'
+import Layout from '../components/Layout'
+import { getContestList } from '../services/reventService'
+import React, { useState, useEffect } from 'react'
+import SearchInput from '../components/SearchInput'
 
-export default class index extends Component {
-  render() {
-    return <div>Home Page</div>
+export default function Home(props) {
+  const { contestList } = props
+
+  const [statusFilter, setStatusFilter] = useState('All')
+
+  const [open, setOpen] = useState(false)
+
+  const filteredContestList = contestList.filter(
+    (contest) => statusFilter === 'All' || contest.status === statusFilter
+  )
+
+  const handleStatusFilter = (status: string) => {
+    setStatusFilter(status)
+  }
+
+  useEffect(() => {
+    // When hiding the filters, reset them
+    if (open == false) {
+      setStatusFilter('All')
+    }
+  }, [open])
+
+  return (
+    <Layout>
+      <div className='p-8 bg-gray-100'>
+        <div className='px-8'>
+          <div className='flex-row'>
+            <button
+              className='text-gray-700 bg-gray-500 text-white px-3 py-2 rounded-2xl font-medium cursor-pointer mr-2'
+              onClick={() => setOpen(!open)}
+            >
+              Filters
+            </button>
+
+            <SearchInput />
+            {open && (
+              <div>
+                <button
+                  className={
+                    statusFilter === 'All'
+                      ? 'text-orange-500 bg-orange-700 text-white px-3 py-2 rounded-2xl font-medium cursor-pointer mr-2 mt-2'
+                      : 'text-gray-700 bg-gray-700 text-white px-3 py-2 rounded-2xl font-medium cursor-pointer mr-2 mt-2'
```

Comment:
since these are used for all the filters you can create a variable for it

Reply:
and you can do the same thing I recommended on the navbar:
```
const filterText = statusFilter === 'Open' ? 'text-orange-500' : 'text-gray-700'
const filterBackground = statusFilter === 'Open' ? 'bg-orange-700' : 'bg-gray-700'
```
since this can change you can use a useState instead for `filterText` and `filterBackground`